pkg/apis/monitoring/v1/prometheus_types.go

```diff
@@ -1848,4 +1848,10 @@ type ScrapeClass struct {
 	//
 	// +optional
 	Relabelings []RelabelConfig `json:"relabelings,omitempty"`
+
+	// SampleLimit defines per-scrape limit on number of scraped samples that will be accepted.
+	// Only valid in Prometheus versions 2.45.0 and newer.
+	//
+	// +optional
```

Comment:
Should we explain the order in which the configuration is generated, similar to what we do with the other ScrapeClass fields?

We have quite a few options to limits now:
* Limits set in .*Monitor objects
* Limits in ScrapeClass, set in Prometheus
* Enforced Limites, set in Prometheus

The order could get very confusing for beginners 😬

Reply:
@ArthurSens can you check it back please?